[EuiBasicTable] Show `nameTooltip` on header cell when it's focusable (#8644)
elastic/eui: packages/website/docs/components/tabular-content/tables/basic.mdx

  - **Note:** the basic table will treat any cells that use a `render` function as being `textOnly: false`. This may cause unnecessary word breaks. Apply `textOnly: true` to ensure it breaks properly.

If a column name needs a longer description, an icon with a tooltip can be used. This can be done using the `columns.nameTooltip` property. This approach ensures that the icon remains visible even if the text gets truncated, unlike manually using a React node for `columns.name`.

> In case the column header cell is sortable, the tooltip will be added on the cell directly instead of the icon to ensure the tooltip is accessible.

Reviewer: **nit-pick:**

```suggestion
If the column header cell is sortable, the tooltip will be applied to the cell itself rather than the icon to ensure accessibility.
```

> In case the column header cell is sortable, the tooltip will be added on the cell directly instead of the icon to ensure the tooltip is accessible.

Reviewer: **nit-pick:**

```suggestion
If the column header cell is sortable, the tooltip will be applied to the cell itself rather than the icon to ensure accessibility.
```

Author: Updated in https://github.com/elastic/eui/pull/8644/commits/9880f8d05995fd4ff4fe714dc10750f0d8e88022
